### ISSUE TITLE
TPB Problem persists

### ISSUE THREAD

**Issue by ricebus** [SiCKRAGETV/sickrage-issues#3117]
SR Version:	BRANCH: (master) / COMMIT: (0b2254c884f833bbb54b8c7e6f6231f4d61557d0)
Python Version:	2.7.10 (default, Sep 26 2015, 00:25:40) [GCC 4.2.3]
pip list output:
```
certifi (2015.9.6.2)
cffi (1.2.1)
cryptography (1.0)
enum34 (1.0.4)
idna (2.0)
ipaddress (1.0.14)
pip (7.1.2)
pyasn1 (0.1.8)
pycparser (2.14)
pyOpenSSL (0.15.1)
requests (2.7.0)
setuptools (18.2)
six (1.9.0)
urllib3 (1.12)
```
I'm running on a QNAP ARM TS-419P II.

```
2015-10-05 17:13:09 FINDPROPERS :: [ThePirateBay] :: Traceback (most recent call last):
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/helpers.py", line 1622, in getURL
    resp = session.get(url, timeout=timeout, allow_redirects=True, verify=session.verify)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/sessions.py", line 476, in get
    return self.request('GET', url, **kwargs)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/sessions.py", line 464, in request
    resp = self.send(prep, **send_kwargs)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/sessions.py", line 576, in send
    r = adapter.send(request, **kwargs)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/cachecontrol/adapter.py", line 46, in send
    resp = super(CacheControlAdapter, self).send(request, **kw)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/adapters.py", line 370, in send
    timeout=timeout
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/packages/urllib3/connectionpool.py", line 544, in urlopen
    body=body, headers=headers)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/packages/urllib3/connectionpool.py", line 344, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/packages/urllib3/connectionpool.py", line 314, in _raise_timeout
    if 'timed out' in str(err) or 'did not complete (read)' in str(err):  # Python 2.6
TypeError: __str__ returned non-string (type Error)

2015-10-05 17:12:53 FINDPROPERS :: [ThePirateBay] :: Unknown exception in getURL https://thepiratebay.gd/s/ Error: __str__ returned non-string (type Error)
```

My log is overwhelmed with these messages, removing 'Verify SSL' option doesn't help (tried removing cache).
Also, I ran `python tests/ssl_sni_tests.py` from the SR  folder and it returned no errors.

**Comment by SiCKRAGEB0T** [comment]
## Thanks for the bug/issue report
Please make sure to submit all of the [required information](https://github.com/SiCKRAGETV/sickrage-issues/blob/master/README.md), or we won't be able to help you!
Provide as **much** information as you can such as debug logs, commit hash, and tell us how to replicate the problem.For basic **questions** use the [ [forum] ](https://sickrage.tv/forums/), and for **Feature Requests** [ [Feathub] ](http://feathub.com/SiCKRAGETV/SickRage)
Also make sure the title accurately describes your issue. ("SR not working" etc. is not ok!)

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist)

## Many issues can simply be solved by:
- Making sure you update to the latest version.
- Have you tried turning your device off and on again?
- Using the **search** function to see if this issue has already been reported/solved.
- Do a [GIT reset](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#update-problems-try-this).
- Checking the [ [Wiki] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki) for [ [Guides] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Installation-&-Configuration-Guides), [ [FAQS] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes) [ [Main settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Settings-explained), [ [Show settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Show-settings-explained) [ [Remaining settings] ](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Remaining-settings-explained)
- Shutting down SickRage and removing the `cache/mako` & `cache/sessions` folders.
- If SickRage doesnt find any episodes you probably need to :
 - add a [Scene Exception](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Scene-exceptions-and-numbering)
 - Set the [Default Episode Status](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#newly-aired-shows-are-not-downloading-and-set-to-skippedignored) to `WANTED`
 - Make sure you understand how [quality detection](https://github.com/SiCKRAGETV/sickrage-issues/wiki/Quality-Settings#quality-detectiondetermination) works during searches.
 - Shutdown SickRage and remove the `cache.db` file.
- If you use a QNAP and have SSL errors try this [package](http://apps.qnap.community/11-community/17-qsickrage).
- If you have a `'ascii' codec can't encode character` error, than set your Locale to [UTF-8](https://github.com/SiCKRAGETV/sickrage-issues/wiki/FAQ%27s-and-Fixes#i-have-problems-with-special-characters-%C3%A9-etc-what-can-i-do)

*ps. If you enjoy Sickrage and like the support provided, than please consider making a [donation](https://github.com/SiCKRAGETV/SickRage/wiki/Donations) to keep SickRage alive.*

**Comment by ricebus** [comment]
I don't understand, how is it related to downloading to season packs? I switched it back to episodes just to make sure. it didn't help.

**Comment by doczong** [comment]
ricebus, I opened a similar issue and @miigotu explained it is the QNAP.  And I am inclined to agree based on his explanation mine was with KAT however, but after looking into my limited log collection, it looks like the error like yours was happening prior to this upgrade but it might have been hidden as there was no "warnings" log selector before.

My biggest question is, I have the same NAS - how in the bloody *($&# did you manage to get these onto your machine?  I can't compile anything.  https://github.com/SiCKRAGETV/sickrage-issues/issues/3099  I'd love some advise to see if I can't assist in tinkering, but I just have been out of compiling on linux for so long that I need a little advice.

Or, if you can share your modules, I would appreciate it as well assuming we're running the same of everything else.

cffi (1.2.1)
cryptography (1.0)
pyOpenSSL (0.15.1)

**Comment by ricebus** [comment]
I actually wanted to wait and try to resolve the issue completely before I release this but as I'm not getting any straight forward answers from @miigotu I'll just share what I have so far.
After endless reading and trying to make the Python package that came with QNAP I understood that the problem wasn't the openssl installation that came with the NAS but rather Python itself. So I compiled the latest Python 2.7.10 with all needed modules. but still... this doesn't solve the whole issue.

You can download my Python QPKG and see if it helps (make sure you delete the old one): https://www.dropbox.com/s/9d39bjp3s2r0xak/Python_2.7.10_arm-x19.qpkg?dl=0
If you're having trouble starting SickRage after the installation try deleting e the entire 'arm' directory that comes with the SickRage QPKG because it has some outdated ssl modules in it (you can live the rar binary tho, cause QNAP delivers an old one)

P.S.
If you look at your issue and mine you can  see that we **don't have the same error messages**. you're is due to lack of SNI support which will be resolved after you install my QPKG. mine is due to an **unknown** reason: @miigotu, any help here?

**Comment by miigotu** [comment]
Patience? When I know something so will you.

**Comment by doczong** [comment]
@ricebus thanks.. will try this out and report back by tomorrow,  You are right the errors are different, I mistakenly looked at both and thought they were the same.  I'll let you know how it goes.

**Comment by doczong** [comment]
@ricebus  Just an update, it's been a long couple days at work.  I apologize for not getting to test the package yet, I am hoping to give it a go tomorrow, or by the weekend anyhow.

**Comment by IronRobi** [comment]
@ricebus your build solved my problem for kat.cr but I still get the Connection error to getURL for TPB. All other errors are gone. So for now I've just disabled TPB since I can live without it.

To get it running I had to delete the arm folder and I also had to update the path for python from usr/bin/python2.7 to opt/bin/python2.7 in the .sh files for SickRage, CouchPotato and Headphones and they all start and run flawlessly. Just wanted to leave a note for all other QNAP arm users that there is hope and to say thanks for the help!

**Comment by fernandog** [comment]
@IronRobi please open a new issue

**Comment by claudiocc** [comment]
@IronRobi Can you put the QPKG available for download? The dropbox link is no longer available.

**Comment by ricebus** [comment]
https://www.dropbox.com/s/7udod90304eh0jt/Python_2.7.10_arm-x19_by_ricebus.qpkg?dl=0

**Comment by miigotu** [comment]
@ricebus you should add it to http://apps.qnap.community/

**Comment by claudiocc** [comment]
@IronRobi Downloaded and tested following your installation instructions. Works like a charm in QNAP Arm. This must be in some way highlighted. I spent more than one hour reading between different closed issues, until reaching here and finally found it.

**Comment by miigotu** [comment]
@neoatomic can you add that link to the wiki when you get a chance, link it in the final post of this issue, and lock it?

**Comment by doczong** [comment]
Hey all, just thought I would chime in, I got around to installing it last friday (two days ago) and I have not yet seen the "Unknown exception in getURL" that @ricebus had been seeing that started this thread.

So, I think this is a solid solution and is better than the QnapWare + Python 2.7.10 + PyOpenSSL-0.13.1 option otherwise available.

**Comment by miigotu** [comment]
If you have python 2.7.9 or 2.7.10 you don't even need pyOpenSSL.

**Comment by neoatomic** [comment]
@miigotu Done. https://github.com/SiCKRAGETV/sickrage-issues/wiki/Sickrage-installation-packages#qnap

@clinton-hall When you have time, could you update your [package](http://apps.qnap.community/11-community/5-sickrage) to include python 2.7.10.? HTTPS is currently not working properly.

**Comment by miigotu** [comment]
@clinton-hall pyOpenSSL isnt even needed at all for SR with Python 2.7.10. Im sure that simplifies your package just a bit.

**Comment by clinton-hall** [comment]
The problem is I don't think python2.7.10 is available on all platforms.
Will need to check... So many different App versions out there.

**Comment by miigotu** [comment]
@clinton-hall the package above is for arm and has 2.7.10 (x86 and x64 were already available)

**Comment by clinton-hall** [comment]
ok, so there are 2 different python Apps, one for arm, and another for intel...
Do these both link Python into the same location? /opt/QPython2

**Comment by doczong** [comment]
Clinton (for ARM):
1) There is Optware which is 2.7.3 in which links to /opt
2) There is the old Python 2.7.3 QPKG which is actually 2.7.0 and overrides Optware and links to /opt
3) There is the new Python 2.7.10 QPKG which overrides anything as well and links to /opt
4) The Qnapware / Entware which to avoid conflicts links to /Apps/opt

All set themselves up in PATH so caling "python" works without hard linking to a specific binary.

The above 2.7.10 package AND the Entware/Qnapware work with SR.

The major problem with both is that you have old libssl/libcrytpo098 binaries in the "/arm/lib" (and "/Repository symlinks) folder of the SickRage QPKG.  Delete those and either of the above start working fine.  

You also have old PyOpenSSL in the /arm/python folder (but your precompiled Cheetah 'C' NameMapper works great).  The /arm/bin folder is essentially redundant if you use Entware/Qnapware and the custom 2.7.10 package above.

The /arm/binutils is fine.  (unrar)

Hope that helps.  I have hack'n'slashed your QPKG wrapper/launcher to nuke the Pyhton "2.7.3" (sic) QPKG check.

The GIT QPKG is out of date and could use an update, but otherwise works fine.

**Comment by miigotu** [comment]
Cheetah is no longer required =P

**Comment by clinton-hall** [comment]
@doczong 

Thanks for the detail. I'll try and update this soon... Can I assume the same is true of x86/64 versions?

Regarding git.. when i built it, it was a LOT more current than anything else on qnap... I can look to update if new features/bug fixes required, but as you said, this works ok...

I myself have tried to avoid Optware as it is dead... and Entware because I don't think it really has a big future... 
As you said, these all put themselves in PATH, but the concern is which one is the first listed? This comes down to order of installation in many cases, hence my previous versions had forced path references/links to try and ensure we used a compatible version...

**Comment by doczong** [comment]
@clinton-hall 
I can't tell you anything about the x86/x64 because from what I understand most of these issues have already been worked around as compilation (arm needs cross compilation) is much easier.

Anything starting with a "Q" in the package name is Intel.

You git is technically older than CVE-2014-9390 which doesn't actually affect linux boxes anyhow, but there is that one vulnerability.

The git that comes with Entware/Qnapware works for git://, except it needs HTTPS and that needs to be compiled or you cant do a git https:// call.  So I am still using your git instead of the much newer one in Entware/Qnapware.  Cross Compilation being the issue.

Perhaps just do a --version on your Package Wrapper when you invoke python2.7?  Just ensure it's 2.7.10 - which would take care of either Qnapware of the new Python QPKG courtesy of @ricebus.  

But yeah, gotta get rid of the old ssl libraries in the arm folder if for no other reason then then screw up SNI.

**Comment by ricebus** [comment]
@doczong I tried to make my package as identical as possible to the one provided by QNAP so that the start script of SR will be fully compatible with it.
Is there really a need for a newer Git package? I can probably make one as well..

**Comment by doczong** [comment]
@ricebus - re Git, nah, not really.  If you google CVE-2014-9390 you will see that there s a vulnerability affecting git, any version under 2.1.0, which the QPKG is.  However, that affects Windows and Mac (any OS that does not differentiate between upper and lower case), so linux is essentially safe.

Should it be updated?  Probably.

Does it need to be updated?  Not really.

**Comment by ricebus** [comment]
The one thing I still don't understand, does TPB work for anyone? or is this a problem for all SR users?

**Comment by miigotu** [comment]
TPB works fine here. It is however blocked by many EU isp/dns.  Working on a feature to allow specifying mirrors rather than the glype proxy soon, because the proxies aren't even working most of the time anymore.

**Comment by clinton-hall** [comment]
Ok, I'll try and get you a new qpkg for sickrage, and I can also compile the latest git to test...

**Comment by ricebus** [comment]
@miigotu well, it works for me when I access it from chrome but it still doesn't work through SR. I get the weird timeout message. It even works with couch potato

**Comment by miigotu** [comment]
@ricebus SNI problem then.

**Comment by ricebus** [comment]
@miigotu so you're telling me even tho I got Python 2.7.10 to work and passed the sni test, it's still an sni problem?

**Comment by miigotu** [comment]
It's possible, there are a lot of moving parts to this. Like for instance if you have pyOpenSSL installed and it is not working correctly (which isnt even needed if you have 2.7.10)

**Comment by doczong** [comment]
@ricebus - my old copy of Python 2.7.10 passed the SNI test scripts as well, but failed in SR.

However, in my case, after nuking the QPKG /arm folder directly, it also started to work fine, except not with the TPB.  KAT magically started working great.  No more SNI errors with KAT after the "arm" folder was gone.  I did still have PyOpenSSL running though.  

I will go back to my (still installed) Qnapware package and test removing PyOpenSSL and see if it works (question: does CP require PyOpenSSL... I know it checks and complained on startup in the logfiles if PyOpenSSL was not 15.1, but I don't know if it actually is *required*).

In my case with the QnapWare version, both CP and SR worked great, after removing the "arm" folder in the QPKG.  Will confirm if TPB works or not (I stopped using it ages ago as noted above).

**Comment by miigotu** [comment]
@doczong if CP warns when you have python 2.7.10 and OpenSSL 1.0.1e+ then you should open an issue with them.

pyOpenSSL is only needed up until Python 2.7.8. Python 2.7.9-2.7.10 built in SSL module can handle SNI

OpenSSL 1.0.1e+ (the c library) is required to properly support the TLS_1.1 and TLS_1.2 encryption methods for some providers (TPB?) that disallow older encryptions.

**Comment by ricebus** [comment]
Well, thanks to @miigotu latest insight on the significance of OpenSSL lib version, I've decided to try and compile Python with newer libs. This time I didn't bother adding all the pyOpenSSL modules etc that I added last time as @miigotu suggested they are irrelevant. 

I can finally say, this issue is resolved! WOOHOO!
TPB works, KAT works, everything works!
I've contacted the guys on app.qnap.community to try and add this package to their library, meanwhile I've uploaded it to my Dropbox.

P.S. I've tested this with CP as well just to make sure I didn't break anything.

https://www.dropbox.com/s/dvnf8r0rfkn0zlp/Python_2.7.10_arm-x19_by_ricebus_v2.qpkg?dl=0

Enjoy!

**Comment by miigotu** [comment]
@clinton-hall @ricebus is there a way to specify qpkg dependancies for other qnap packages? It would seem more sane to "require" the proper Python qpkg and have an architecture independant SickRage package...

**Comment by ricebus** [comment]
Yes, you can check the QDK guide, page 5 (look for "QPKG_REQUIRE") http://download.qnap.com/dev/QDK_2.0.pdf

**Comment by rascalli** [comment]
I installed this package , but I still get : 

https://kat.cr/tv/?q=&field=time_add&category=tv&sorder=desc&rss=1 Error: "'_socketobject' object has no attribute 'set_tlsext_host_name'"


What I did was : 

1. upgrade python with the ARM package
2. Restart NAS
3. Restart SR

Any idea if I mised something ?

**Comment by ricebus** [comment]
Did you delete the ssl libs in the arm folder in the Sr package?

**Comment by doczong** [comment]
@ricebus 
I am so behind... I haven't even got to testing the Entware/Qnapware with removing PyOpenSSL yet (but I did open a ticket on CP to see if it is actually required, based on @miigotu 's comments earlier re requiring Python 2.7.9-2.7.10 and OpenSSL 1.0.1e+  or higher).

Can you give a package list what you have installed with Python?  Did you need PyASN1 still, as well as ntd-httpsclient?  Or are you basically Python 2.7.10 with OpenSSL 1.0.1e+ or higher?

I'd like to see if I can't replicate that in the Entware/Qnapware as well to give us a 2nd option!

**Comment by rascalli** [comment]
@ricebus , what folder are they in  ? (don't want to delete something I need)

**Comment by ricebus** [comment]
/share/MD0_DATA/.qpkg/SickBeard-TVRage/arm/

you can delete everything in that folder except for bin/unrar

**Comment by eraldog** [comment]
Hi all
I'm trying to fix this problem on my Qnap ts-221 and I'm looking for a solutions for days.
I installed the last Python 2.7.3 from the official Qnap qpkg and I have SickRage updated by the last master commit.
Now I installed the package from @ricebus wihout luck, but the strange thing is:
```
[~] # openssl version
OpenSSL 1.0.1p 9 Jul 2015

[~] # python --version
Python 2.7.3

[~] # python -c "import sys; import os; import ssl; print(ssl.OPENSSL_VERSION);"
OpenSSL 0.9.8v 19 Apr 2012

[~] # python2.7 --version
Python 2.7.10

[~] # python2.7 -c "import sys; import os; import ssl; print(ssl.OPENSSL_VERSION);"
OpenSSL 1.0.1p 9 Jul 2015

[~] # python2 --version
Python 2.7.5

[~] # /opt/bin/python2.7 --version [this is in sickbeard-tvrage.sh]
Python 2.7.3

[~] # /opt/bin/python2.7 -c "import sys; import os; import ssl; print(ssl.OPENSSL_VERSION);"
OpenSSL 0.9.8v 19 Apr 2012
```

And I don't have any arm folder in /share/MD0_DATA/.qpkg/SickBeard-TVRage.
How can I convince Python to use the correct OpenSSL?
Can I just edit the 6th line in the sickbeard-tvrage.sh from:
DAEMON=/opt/bin/python2.7
to:
DAEMON=python2.7


PS.
This is the error I get in SickBeard:
```
AASSLError: [Errno bad handshake] [('SSL routines', 'SSL23_GET_SERVER_HELLO', 'sslv3 alert handshake failure')]
AA    raise SSLError(e, request=request)
AA  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/lib/requests/adapters.py", line 431, in send
AA    resp = super(CacheControlAdapter, self).send(request, **kw)
AA  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/lib/cachecontrol/adapter.py", line 46, in send
AA    r = adapter.send(request, **kwargs)
AA  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/lib/requests/sessions.py", line 576, in send
AA    resp = self.send(prep, **send_kwargs)
AA  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/lib/requests/sessions.py", line 464, in request
AA    return self.request('GET', url, **kwargs)
AA  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/lib/requests/sessions.py", line 476, in get
AA    resp = session.get(url, timeout=timeout, allow_redirects=True, verify=session.verify)
AA  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/sickbeard/helpers.py", line 1599, in getURL
AATraceback (most recent call last):
2015-10-28 21:04:48 ERROR    SEARCHQUEUE-MANUAL-272644 :: [ThePirateBay] :: Connection error to getURL https://pirateproxy.la/s/?q=12+Monkeys+S01E08&orderby=7&category=200&type=search&page=0 Error: u"error [Errno bad handshake] [('SSL routines', 'SSL23_GET_SERVER_HELLO', 'sslv3 alert handshake failure')]"
2015-10-28 21:04:48 INFO     SEARCHQUEUE-MANUAL-272644 :: [ThePirateBay] :: Performing episode search for 12 Monkeys
```

and this is from CouchPotato2:
```
[chpotato.core._base._core] OpenSSL installed but 0.15 is needed while 0.13.1 is installed. Run `pip install pyopenssl --upgrade`
10-28 20:11:27 DEBUG[chpotato.core._base._core] OpenSSL detected: pyopenssl (0.13.1) using OpenSSL (OpenSSL 0.9.8v 19 Apr 2012)
```

Thank you!

**Comment by ricebus** [comment]
Delete the official Python package, then disable and re-enable my package.

**Comment by eraldog** [comment]
Thanks @ricebus but I a bit noob..
How I delete the official Python package?

**Comment by ricebus** [comment]
Go into the Nas administration panel => app center => select the relevant package and you'll have a removal button.

**Comment by eraldog** [comment]
Sam error and same output for the -- version commands. I also rebboted the Nas.
This is what I have in /share/MD0_DATA/.qpkg/Python/bin
```
-rwxr-xr-x    1 admin    administ       132 Sep 25 23:48 2to3*
-rwxr-xr-x    1 admin    administ       130 Sep 25 23:48 idle*
-rwxr-xr-x    1 admin    administ       115 Sep 25 23:48 pydoc*
lrwxrwxrwx    1 admin    administ         7 Oct 29 17:28 python -> python2*
lrwxrwxrwx    1 admin    administ        14 Oct 29 17:28 python-config -> python2-config*
lrwxrwxrwx    1 admin    administ         9 Oct 29 17:28 python2 -> python2.7*
lrwxrwxrwx    1 admin    administ        16 Oct 29 17:28 python2-config -> python2.7-config*
-rwxr-xr-x    1 admin    administ   4551527 Oct 26 22:05 python2.7*
-rwxr-xr-x    1 admin    administ      1718 Oct 26 22:06 python2.7-config*
-rwxr-xr-x    1 admin    administ     18578 Sep 25 23:48 smtpd.py*
```

And in /share/MD0_DATA/.qpkg/Optware/bin
```
-rwxr-xr-x    1 admin    administ        78 Apr 12  2012 pydoc-2.6*
-rwxr-xr-x    1 admin    administ        78 Apr 12  2012 pydoc-2.7*
-rwxr-xr-x    1 admin    administ      3368 Apr 12  2012 python2*
-rwxr-xr-x    1 admin    administ      1618 Apr 12  2012 python2-config*
-rwxr-xr-x    1 admin    administ      3368 Apr 12  2012 python2.6*
-rwxr-xr-x    1 admin    administ      1418 Apr 12  2012 python2.6-config*
-rwxr-xr-x    1 admin    administ      3368 Apr 12  2012 python2.7*
-rwxr-xr-x    1 admin    administ      1618 Apr 12  2012 python2.7-config*
```
Please notice the files from 2012..
It seems SickRage is still using Python 2.7.3 from the Optware/bin folder

**Comment by ricebus** [comment]
I'd suggest you to delete it, you can use the terminal or do it through the Web administration of optware

**Comment by eraldog** [comment]
Do you seggest to delete all of python* files in /share/MD0_DATA/.qpkg/Optware/bin folder?
Is it safe? For the system and the other app?
Or maybe can I also run "ipkg remove python27" ?

**Comment by doczong** [comment]
Eraldog: I documented this on the Qnap Forum for SR (sort of).

If you have the old Optware, delete those Symlinks in /opt/bin to python.  Or, delete the old Python via ipkg or ipkg-web.  Once you do, those symlinks will go away.  I am not certain, but I beleive those symlinks are created each time Optware runs (launches... at startup).

You could also, if you are just testing things, modify the .qpkg/Sickbeard-TVRage/sickbeard.sh statup script... about 10 lines from the top, you can change specifically to a different python by changing the path there as well, saving it, and reloading SR.

**Comment by ricebus** [comment]
running `ipkg remove python27` is the right way

**Comment by eraldog** [comment]
@doczong Can you give me a link? I didn't find your docs. Thanks

I did `ipkg remove python27` then from Qnap app web management:
stopped SickRage
stopped Couchpotato
stopped Python 2.7.10
started Python 2.7.10
started SickRage
started Couchpotato

Both SR and CP failed to start: 
`Failed to start SickBeard-TVRage, /opt/bin/python2.7 was not found. Please re-install the Pythton ipkg.	`

Then I changed the line in `sickbeard-tvrage.sh` and `CouchPotato2.sh`
DAEMON=/opt/bin/python2.7
to:
DAEMON=/share/MD0_DATA/.qpkg/Python/bin/python2.7

Restarted SR and now it's working without errors with PB!

But now CP complains about OpenSSL...
```
10-29 19:00:12 ERROR [31m[chpotato.core._base._core] OpenSSL not available, please install for better requests validation: `https://pyopenssl.readthedocs.org/en/latest/install.html`: Traceback (most recent call last):
  File "/share/MD0_DATA/.qpkg/CouchPotato2/couchpotato/core/_base/_core.py", line 84, in dependencies
    import OpenSSL
ImportError: No module named OpenSSL
```
Do I have to change that line after every SR/CP updates?

**Comment by ricebus** [comment]
no, you dont. btw mine is set to /usr/bin/python.
maybe you're using an old CP package? mine is V141108 by clinton.hall.

**Comment by doczong** [comment]
@eraldog 
But now CP complains about OpenSSL...

Just a longshot...

Python 2.7.10 will need the new OpenSSL.... what Firmware version are you running?  I don't know anything about 4.x, but if you are running 3.8.3 (or thereabouts) you would have way too old of an OpenSSL version.  That firmware comes with OpenSSL 0.9.8v2.

The OpenSSL in Optware is also still too old.

If you are adverse to upgrading your firmware like I am, you should install QnapWare and replace all your packages you have in Optware with its packages.  It comes with a new OpenSSL 1.0.2d which should satisfy @ricebus Python 2.7.10 version 2.0 package (which does not have PyOpenSSL) in it any longer.

Which is why I have not got around to his Python package and am instead using Qnapware, as I refuse to upgrade the firmware at the moment and it has all the newer packages I need (gep, sed, etc).

@ricebus... when you precompiled your new Python... did you include PyASN1 and NTD-HTTPSCLIENT (I note you did not have the latter in your first package but did have PyASN1).  I'm going to try and get around to removing PyOpenSSL-0.13.1 this weekend from my otherwise working Qnapware as an option for those on older Firmwares.

**Comment by ricebus** [comment]
I can make you guys an OpenSSL package if you'd like. but I don't know if it'll help.
any takers?

**Comment by doczong** [comment]
@ricebus - can you provide a python package list that you have in yours like you have in the OP?  Please and thanks.

**Comment by eraldog** [comment]
@ricebus /usr/bin/python in my system point to that I changed in my .sh files:
`/share/MD0_DATA/.qpkg/Python/bin/python2.7`  so it's the same. 

My CP package from QNAP web management is V130805 by clinton.hall but the app version is e6fa8b8b (4/10/2015 11:13:08) Updater git, master.
Do you think it's too old? Can I install a newer one over my package?

@doczong I'm on the last firmware 4.2.0 build 20150925.
So it should contain a newer OpenSSL tan 0.9 I guess..

**Comment by ricebus** [comment]
@eraldog try update the package first.
@doczong no extra packages at all.

**Comment by doczong** [comment]
@ricebus so just python 2.7.10, and nothing else comes with?

ie: None of what you had above?  (if you are working with just that then I'll try it without ntd-httpsclient and also pyasn1 as well tomorrow/saturday)

certifi (2015.9.6.2)
cffi (1.2.1)
cryptography (1.0)
enum34 (1.0.4)
idna (2.0)
ipaddress (1.0.14)
pip (7.1.2)
pyasn1 (0.1.8)
pycparser (2.14)
pyOpenSSL (0.15.1)
requests (2.7.0)
setuptools (18.2)
six (1.9.0)
urllib3 (1.12)

**Comment by ricebus** [comment]
yup, they have all the required modules included with SR already.

you can also try this openssl qpkg
https://www.dropbox.com/s/drmlt9w3ukftge8/OpenSSL_1.0.1p_arm-x19_by_ricebus.qpkg?dl=0

**Comment by eraldog** [comment]
@ricebus I upgraded the CP package but nothing changed.
I tried to use pip to check the pyopenssl version but pip gave an error:
```
# pip list
Traceback (most recent call last):
  File "/usr/bin/pip", line 7, in <module>
    from pip import main
ImportError: No module named pip
```
Maybe because I had pip in the old Python 2.7.3 packages.
So i installed with your 2.7.10 and I added pyopenssl 0.13.1  --> CP same error
Then I tried to upgrade pyopenssl but fails with many gcc errors on the dependancies.

CP error now:
```
10-30 00:57:14 ERROR [31m[chpotato.core._base._core] OpenSSL not available, please install for better requests validation: `https://pyopenssl.readthedocs.org/en/latest/install.html`: Traceback (most recent call last):
  File "/share/MD0_DATA/.qpkg/CouchPotato2/CouchPotato2/couchpotato/core/_base/_core.py", line 84, in dependencies
    import OpenSSL
  File "/share/MD0_DATA/.qpkg/Python/lib/python2.7/site-packages/OpenSSL/__init__.py", line 36, in <module>
    from OpenSSL import crypto
ImportError: libssl.so.0.9.8: cannot open shared object file: No such file or directory
```
Result from `pip list`:
```
pip (7.1.2)
pyOpenSSL (0.13.1)
setuptools (18.4)
```

**Comment by miigotu** [comment]
@eraldog you need to REMOVE pyOpenSSL. pyOpenSSL is not the same thing as OpenSSL.

**Comment by doczong** [comment]
FYI guys, removing PyOpenSSL from Python 2.7.10 under Qnapware/Entware also works perfectly.  KAT working, TPB working, no SSL warnings or errors.  CP also working and snatching and DL'ing.  Finally got around to testing.

**Comment by eraldog** [comment]
@miigotu Ok I removed pyOpenSSL previosly installed via pip under @ricebus Python2.7.10 packages.
I have:
`openssl version` --> OpenSSL 1.0.1p 9 Jul 2015
`which openssl` --> /usr/bin/openssl
So OpenSSL is on my system but CP give the same error:
```
OpenSSL not available, please install for better requests validation: `https://pyopenssl.readthedocs.org/en/latest/install.html`: Traceback (most recent call last):
  File "/share/MD0_DATA/.qpkg/CouchPotato2/CouchPotato2/couchpotato/core/_base/_core.py", line 84, in dependencies
    import OpenSSL
ImportError: No module named OpenSSL
```
Maybe is it looking for OpenSSL inside the python2.7.10?
Do I have to remove the pyOpenSSL that come with @ricebus package too?
How can I do?

Thanks and sorry I very noob at this

**Comment by clinton-hall** [comment]
Sorry for the lengthy delay here...
I have a new version of SickRage qpkg to test.
https://www.dropbox.com/s/yiqj5a7v4jp32wi/SickBeard-TVRage_151107.qpkg

Please let me know if this works as needed, and I'll update the post at QNAP forums.

basically, this tests all likely Python install locations to find the latest version, then Starts SickRage with that version.
I have stripped out all libs and python modules etc.
I haven't played with the qpkg dependency checks (i.e I believe this still required "python" qpkg, so please let me know if this needs to change.

**Comment by yelpspoon** [comment]
Hi all.
I have a QNAP TS-420(ARM).
I was experiencing the 'See: http://git.io/vJrkM' with KAT and my other torrent index.
I wiped caches, unchecked SSL in SR etc.  with no successful KAT search.

I (think I) read through all of this thread (and some related threads) and done the following:
Removed (via QNAP WebGUI) Py 2.7.3.  I installed ricebus' Python 2.7.10 (though python2.7 --version shows 2.7.5) and installed clinton-hall's SR 151107 and still get the above error with and additional SSL error:
AASSLError: [Errno 1] _ssl.c:504: error:14077438:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert internal error
AA    raise SSLError(e, request=request)


Additional library/bin info (btw, the system looks a mess):

[~] # openssl version
OpenSSL 1.0.1p 9 Jul 2015

[~] # python --version
Python 2.7.3

[~] # python -c "import sys; import os; import ssl; print(ssl.OPENSSL_VERSION);"
OpenSSL 0.9.8v 19 Apr 2012

[~] # python2.7 --version
Python 2.7.10

[~] # python2.7 -c "import sys; import os; import ssl; print(ssl.OPENSSL_VERSION);"
OpenSSL 1.0.1p 9 Jul 2015

[~] # python2 --version
Python 2.7.5

[~] # /opt/bin/python2.7 --version [this is in sickbeard-tvrage.sh]
Python 2.7.3

[~] # /opt/bin/python2.7 -c "import sys; import os; import ssl; print(ssl.OPENSSL_VERSION);"
OpenSSL 0.9.8v 19 Apr 2012

**Comment by clinton-hall** [comment]
To check my new qpkg is using the correct Python, can you show the output of

   /etc/init.d/sickbeard-tvrage.sh restart

Also, show the output of

  which python2.7

**Comment by yelpspoon** [comment]
@clinton-hall (thanks)
Found Python Version 2.7.3 at /opt/bin/python2.7                                                                                     
Updating SickBeard-TVRage                                                                                                            
HEAD is now at c195264 Fixes https://github.com/SiCKRAGETV/sickrage-issues/issues/3580                                               
Already up-to-date.     

which python2.7
/usr/local/bin/python2.7

**Comment by felipesalomao** [comment]
Hi, i am getting bellow error:
2015-11-11 07:18:23 ERROR    SEARCHQUEUE-MANUAL-274431 :: [KickAssTorrents] :: Connection error to getURL https://kickass.unblocked.la/usearch/?q=Gotham+S02E07&field=seeders&rss=1&category=tv&sorder=desc Error: u'See: http://git.io/vJrkM'

2015-11-11 07:18:14 ERROR    SEARCHQUEUE-MANUAL-274431 :: [ThePirateBay] :: Connection error to getURL https://pirateproxy.la/s/?q=Gotham+S02E07&orderby=7&category=200&type=search&page=0 Error: u'See: http://git.io/vJrkM'

I know that is big ssl problems.. i got this error some months ago, then disabled thepiratebay and kickass worked fine without error.. Now kickass also have this SSl problem.. I tried all to solve it.. 

i Have installed: 
Qnapware 0.90
optware IPKG 0.99.163
Openssl 1.0.1p 
sickbeard-tvrage 151107

**Comment by felipesalomao** [comment]
I am getting errors:
2015-11-11 07:29:20 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [KickAssTorrents] :: Connection error to getURL https://kickass.unblocked.la/anime/?q=&field=time_add&rss=1&category=tv&sorder=desc Error: u'See: http://git.io/vJrkM'
AASSLError: [Errno 1] _ssl.c:504: error:14077438:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert internal error

2015-11-11 07:29:19 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [KickAssTorrents] :: Connection error to getURL https://kickass.unblocked.la/tv/?q=&field=time_add&rss=1&category=tv&sorder=desc Error: u'See: http://git.io/vJrkM'

I tried all to solve it, reinstalled sickrage, installed qnapware, openssl, optware, etc.. How solve it ? 
I already had issue with ssl for thepiratebay some monhts ago, then enabled kickass and it worked great, but now i also have ssl problem with kickass and i am not able to download anymore..

**Comment by clinton-hall** [comment]
@yelpspoon @felipesalomao 

can you please test this latest version?
https://www.dropbox.com/s/8aii7d98oun4x2h/SickBeard-TVRage_151111.qpkg

And ensure you have the latest Python 2.7.10 for your NAS.

**Comment by yelpspoon** [comment]
@clinton-hall 
Will-do.
As for the py2.7.10; how would I ensure other than ricebus' qpkg?

Regards

**Comment by yelpspoon** [comment]
@clinton-hall 
Same results:
2015-11-11 10:58:35 ERROR    SEARCHQUEUE-MANUAL-193131 :: [KickAssTorrents] :: Connection error to getURL https://kickass.unblocked.la/anime/?q=&field=time_add&rss=1&category=tv&sorder=desc Error: u'See: http://git.io/vJrkM'

:/

**Comment by clinton-hall** [comment]
Ok... not sure if there is an issue with the Python modules built in to the QPython2.7.10 App or SickRage...

When using /opt/QPython2/bin/python2.7 on QNAP X86
```
Traceback (most recent call last):
  File "SickBeard.py", line 58, in <module>
    import sickbeard
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/__init__.py", line 36, in <module>
    from sickbeard import metadata
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/metadata/__init__.py", line 22, in <module>
    from sickbeard.metadata import kodi, kodi_12plus, mediabrowser, ps3, wdtv, tivo, mede8er, generic, helpers
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/metadata/kodi.py", line 21, in <module>
    from sickbeard.metadata import generic
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/metadata/generic.py", line 30, in <module>
    from sickbeard import helpers
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/sickbeard/helpers.py", line 44, in <module>
    import adba
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/adba/__init__.py", line 24, in <module>
    from aniDBAbstracter import Anime, Episode
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/adba/aniDBAbstracter.py", line 23, in <module>
    import aniDBfileInfo as fileInfo
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/adba/aniDBfileInfo.py", line 30, in <module>
    import requests
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/__init__.py", line 53, in <module>
    from .packages.urllib3.contrib import pyopenssl
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/packages/__init__.py", line 75, in load_module
    __import__(name)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/requests/packages/urllib3/contrib/pyopenssl.py", line 50, in <module>
    from ndg.httpsclient.ssl_peer_verification import SUBJ_ALT_NAME_SUPPORT
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/ndg/__init__.py", line 19, in <module>
    __import__('pkg_resources').declare_namespace(__name__)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 2627, in <module>
    add_activation_listener(lambda dist: dist.activate())
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 683, in subscribe
    callback(dist)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 2627, in <lambda>
    add_activation_listener(lambda dist: dist.activate())
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 2176, in activate
    map(declare_namespace, self._get_metadata('namespace_packages.txt'))
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 1794, in declare_namespace
    _handle_ns(packageName, path_item)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 1765, in _handle_ns
    loader.load_module(packageName); module.__path__ = path
  File "/opt/QPython2/lib/python2.7/pkgutil.py", line 246, in load_module
    mod = imp.load_module(fullname, self.file, self.filename, self.etc)
  File "/share/MD0_DATA/.qpkg/QPython2/lib/python2.7/site-packages/zopeskel/__init__.py", line 2, in <module>
    from zopeskel.basic_namespace import BasicNamespace
  File "/share/MD0_DATA/.qpkg/QPython2/lib/python2.7/site-packages/zopeskel/basic_namespace.py", line 2, in <module>
    from zopeskel.vars import var, DottedVar, StringVar, BooleanVar, TextVar
  File "/share/MD0_DATA/.qpkg/QPython2/lib/python2.7/site-packages/zopeskel/vars.py", line 2, in <module>
    from paste.script.templates import var as base_var
  File "/opt/QPython2/lib/python2.7/site-packages/paste/script/templates.py", line 8, in <module>
    from . import command
  File "/opt/QPython2/lib/python2.7/site-packages/paste/script/command.py", line 54, in <module>
    dist = pkg_resources.get_distribution('PasteScript')
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 315, in get_distribution
    if isinstance(dist,Requirement): dist = get_provider(dist)
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 197, in get_provider
    return working_set.find(moduleOrReq) or require(str(moduleOrReq))[0]
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 670, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py", line 569, in resolve
    raise DistributionNotFound(req)  # XXX put more info here
pkg_resources.DistributionNotFound: PasteScript
```

**Comment by clinton-hall** [comment]
Interesting, just tried with `/share/MD0_DATA/.qpkg/SickBeard-TVRage/SickBeard-TVRage/lib/pkg_resources.py` removed and everything seems to run ok...

**Comment by yelpspoon** [comment]
@clinton-hall 
When you say "everything" do you mean, the "pkg_resources.DistributionNotFound" error you posted above OR the "See: http://git.io/vJrkM" (SSL) error?

**Comment by clinton-hall** [comment]
Sorry, should have been clearer.
With the pkg_resources file present, SickRage will not start. With this file removed, it starts and all basic functions appear to run. I have not specifically tested the SSL, but I was of the understanding that puthon2.7.10 was supposed to work.

The problem with QNAP now is that there are so many different platforms that all have different App versions and firmware. The "fix" suggested didn't work on my NAS, but I am trying to make my App "universal".

At this stage I am trying to find a way to get SR running with "QPython2" without the need to remove/edit SR files.

**Comment by forkless** [comment]
@ricebus 

Thanks for your QPKG, I have added it to the repository. It can be found at 

http://apps.qnap.community/11-community/20-python-community-2

**Comment by bbohling** [comment]
@ricebus: what about us on x86-based qnaps? :)

**Comment by ricebus** [comment]
@bbohling unfortunately I don't own an x86 nas so... I can't really help you guys. BUT! there's the QPython2 QPKG that provides python 2.7.10. the only problem with this pkg is that it's huge (like 1GB) because it contains every module on the planet :P

**Comment by bbohling** [comment]
@ricebus: is it safe to assume it is not as easy as Remove the Python (2.7.3) qpkg from the App Center and then manually installing QPython2? Reading through tons of GitHub issues and forum posts, it will also require some editing of config files??? Sorry, I learned enough to get everything set up once on my QNAP, but that's about it. :(

**Comment by ricebus** [comment]
probably a question you should forward to @clinton-hall as he maintains the SR pkg.
If the SR QPKG can locate QPython2's path, then you're all good.

**Comment by clinton-hall** [comment]
You can use Python qpkg AND QPython2 qpkg.
My new App will check for the latest Python version, but there are issues to be investigated.
What model NAS do you have?

**Comment by bbohling** [comment]
maybe i am unique, but sickrage seems to break quite frequently in the past few months. last night switched to sonarr (took a whopping 5 minutes to completely set up). may be an alternative for those having issues getting sickrage to work. unfortunately i think this is more of a python (and it's dependencies) issue...rather than sickrage. time will tell if i will run into the same situation with sonarr (.net based; via mono)

**Comment by miigotu** [comment]
Kind of shitty of you to advertise a competing application here dont you think?

**Comment by bbohling** [comment]
@milgotu: i did debate my comment. however, given the nature of open source, i thought it was fair. in addition, i called out the issue is likely with python. and there are TONS of people having issues so knowing about alternatives is fair game IMO. if others disagree, i'm willing to retract as i'm just trying to help others...not offend.

**Comment by miigotu** [comment]
It's quintessential to jumping on https://bugzilla.kernel.org/ and saying how you switched to windows and it solved all of your problems.

**Comment by bbohling** [comment]
fair dinkum

**Comment by claudiocc** [comment]
Same here. TPB/KA stop working since last update. Any expected fix?

**Comment by felipesalomao** [comment]
Here, same thing.. Since this update, not working on qnap 219P II
There is no solution yet ?

**Comment by shiftux** [comment]
Many thanks to @ricebus! - may a 1000 virgins serve you in eternety!

Sickrage works like a charm now on my ARM based QNAP (TS220). However the new Python messed with my CouchPotato... I get 
<code> ImportError: No module named __ future __ </code>
It dosn't find any of the python modules anymore. 

would appreciate a hint on this...
thanks in advance!

**Comment by miigotu** [comment]
The url problems are all fixed in https://github.com/SickRage/SickRage
Just hoping someone can update or add the new Org/Repo to the qnap installers soon!

**Comment by ricebus** [comment]
@miigotu is there a manual way to change that?

**Comment by miigotu** [comment]
I'm not sure how the paths are set up with git and SR source on qnap, so I have no clue how to guide you really. Thats why I am hoping the qnap packages are updated.

Can you specify a URL when installing the QPkg?

Harder way
basically:
enter the dir where SickBeard.py is and run:
```bash
/path/to/git remote set-url origin https://github.com/SickRage/SickRage.git
/path/to/git fetch origin
/path/to/git reset --hard origin/master
```

**Comment by doczong** [comment]
@miigotu @ricebus 
That's your "new" repository @miigotu - the one which introduced a lot of these issues since May of 2015.  You shouldn't be trying to get people to come over to your new, same named fork.

**Comment by miigotu** [comment]
Mine? https://github.com/orgs/SickRage/people

Seriously, don't listen to the bs.

**Comment by doczong** [comment]
@miigotu @ricebus @clinton-hall @echel0n 
I'm not listening to BS.  I've read all the facts and have came up with my own opinion.

(also to note, I don't have any of these issues you all are having with KAT using the Entware-ng Python  which... I am still on because I haven't had time to try the new Python package)

I've read all the IRC chat transcripts (and I suggest others do as well - linked on your site, they are long I will say that).  I've read all the hoopla about donations (and I won't comment on that as I don't know where the money flows), I've read all the comments about you guys keeping the same name even though you forked (which, you did fork, you were kicked from the original repo, so yours IS a fork no matter what word games you want to play)....  the fact you won't rename is sad.  

I saw the hack of sickrage.tv 

You and I have worked on this and other similar issues, and I think we have a mutual respect, but please, some people out there actually read things and the games right now are not good for open source.  Rename and move on.   Yours is a fork.  In the true nature of open source, forks rename and move on.  You're not playing by the intent of letter of open source.

You're now coming over to this repo and telling people to come to your similarly named repo.

Simply not cool.  Like, really not cool.  As in, I have no horse in this race but this kind of crap pisses me right off.

**Comment by IronRobi** [comment]
@ricebus I edited the .sh file (the same one I edited for python) and changed the git URL to the one @miigotu posted. Restarted sickrage and it worked fine. Solved my ASCII errors, post processing works again and so does kat.cr. Still get connection issues with TPB although I haven't tried playing around with the custom address setting for it yet. Kat.cr and post processing was most important to me.

With the new git, I do get a warning when I start saying git is not configured properly. I'm not at my system so I can't post the exact warning, but when I go to the git settings under general settings I don't have the option to select master and automatic updates are not working so I had to apply the last update manually.

**Comment by doczong** [comment]
BTW is anyone wants to fix KAT, for some reason the maintainers at the time that got kicked changed sickbeard/providers/kat.py from the kickass URL (kat.cr) to theincorrect https://kickass.unblocked.la

Change the following code in that file to:
'base_url': 'https://kat.cr/',
'search': 'https://kat.cr/%s/',

**Comment by labrys** [comment]
Some people had been having issues with the other.  We have since switched back and now added a custom field for those who the default doesn't work for (i.e. its being blocked by providers etc.)

**Comment by miigotu** [comment]
@IronRobi try re-entering your git username/password to get rid of that warning

@doczong You are welcome to your own opinion, but you dont need to press it upon other people.

**Comment by doczong** [comment]
@miigotu 
In all due respect, I don't live under fascism, and the fact that "my opinion" which is held by others, and which is doing nothing more that simply telling people to read your own published materials before jumping to conclusions, and I surely am not pressing that opinion upon others.  I am simply pointing out a fact that this is still the current repo, and your fork is a different repo, and you trying to get others to mosey on over to yours.

People are free to take my own comments, compare them to your stated comments and make their own opinion.  Trying to whitewash, or worse what almost seems like trying to censor something is also very not cool in my books.

As I noted earlier, I've been dealing with open source since I received my first copy of slackware on cd in the early 90s.  While I don't do a lot of coding anymore, doesn't mean I wasn't active at one point, nor does it discount my experience.

**Comment by miigotu** [comment]
SickRage was a fork to begin with. A project does not reside by it's location, but with its community of users and developers. 15/16 developers have moved to a new location, and well over half of the users.
We have nearly 1000 new commits since we created our new organization.

Im not whitewashing anything, I simply refuse to stoop to the tactics that others have used to tell the story through lies and slander. I'm not here to talk bad about anyone or anything.

You can argue until you are blue in the face, but in the end, the users will decide based on updates and stability.

**Comment by ricebus** [comment]
https://github.com/SiCKRAGETV/SickRage/graphs/contributors

@doczong looks like @miigotu  and the others ARE the main contributors and maintainers of the project. so..

**Comment by doczong** [comment]
@ricebus, you are 100% correct - in 2015.
Go back to 2014 (any time prior to March of 2015 actually) That tells a different story.  Also, not sure if you are aware, but the other guy had a family emergency and they started contributing more and more and now he doesn't and here we are.

Now... go read the IRC chat log having seen the above.

**Comment by labrys** [comment]
Please keep these issue-requests on topic.  Further discussion can be moved to SickRage/sickrage-issues#3 or, if you prefer to discuss it on this repo, create a new issue here at [SiCKRAGETV/sickrage-issues](https://github.com/SiCKRAGETV/sickrage-issues).
